src/sensor_ads1220.c

```diff
@@ -0,0 +1,206 @@
+// Support for ADS1220 ADC Chip
+//
+// Copyright (C) 2024 Gareth Farrington <gareth@waves.ky>
+//
+// This file may be distributed under the terms of the GNU GPLv3 license.
+
+#include "board/irq.h" // irq_disable
+#include "board/gpio.h" // gpio_out_write
+#include "board/misc.h" // timer_read_time
+#include "basecmd.h" // oid_alloc
+#include "command.h" // DECL_COMMAND
+#include "sched.h" // sched_add_timer
+#include "sensor_bulk.h" // sensor_bulk_report
+#include "spicmds.h" // spidev_transfer
+
+struct ads1220_adc {
+    struct timer timer;
+    uint32_t rest_ticks;
+    struct gpio_in data_ready;
+    struct spidev_s *spi;
+    uint8_t flags, data_count;
+    struct sensor_bulk sb;
+};
+
+// Flag types
+enum {
+    FLAG_PENDING = 1 << 0
+};
+
+#define BYTES_PER_SAMPLE 4
+
+static struct task_wake wake_ads1220;
+
+/****************************************************************
+ * ADS1220 Sensor Support
+ ****************************************************************/
+#define MAX_SPI_READ_TIME timer_from_us(150)
+
+static inline uint8_t
+is_flag_set(const uint8_t mask, struct ads1220_adc *ads1220)
+{
+    return !!(mask & ads1220->flags);
+}
+
+static inline void
+set_flag(uint8_t mask, struct ads1220_adc *ads1220)
+{
+    ads1220->flags |= mask;
+}
+
+static inline void
+clear_flag(uint8_t mask, struct ads1220_adc *ads1220)
+{
+    ads1220->flags &= ~mask;
+}
+
+// Event handler that wakes wake_ads1220() periodically
+static uint_fast8_t
+ads1220_event(struct timer *timer)
+{
+    struct ads1220_adc *ads1220 = container_of(timer, struct ads1220_adc,
+                                                timer);
+    set_flag(FLAG_PENDING, ads1220);
+    sched_wake_task(&wake_ads1220);
+    return SF_DONE;
+}
+
+// Helper code to reschedule the ads1220_event() timer
+static void
+ads1220_reschedule_timer(struct ads1220_adc *ads1220)
+{
+    irq_disable();
+    set_flag(FLAG_PENDING, ads1220);
```

Comment:
This code doesn't look right.  It's not clear to me what this flag is intended to do.  If it is a port from sensor_adxl345.c, then it should be clearing the flag here.

Reply:
Resolved to work like ADXL345